tests/bot/cogs/moderation/test_utils.py

```diff
@@ -0,0 +1,411 @@
+                        type="Mute",
+                        expires="2020-02-26 09:20 (23 hours and 59 minutes)",
+                        reason="Test"
+                    ), width=2048, placeholder="..."),
+                    colour=Colours.soft_red,
+                    url=utils.RULES_URL
+                ).set_author(
+                    name=utils.INFRACTION_AUTHOR_NAME,
+                    url=utils.RULES_URL,
+                    icon_url=Icons.defcon_denied
+                ).set_footer(text=utils.INFRACTION_APPEAL_FOOTER),
+                "send_result": False
+            },
+            {
+                "args": (self.user, "mute", None, "foo bar" * 4000, Icons.defcon_denied),
+                "expected_output": Embed(
+                    title=utils.INFRACTION_TITLE,
+                    description=textwrap.shorten(utils.INFRACTION_DESCRIPTION_TEMPLATE.format(
+                        type="Mute",
+                        expires="N/A",
+                        reason="foo bar" * 4000
+                    ), width=2048, placeholder="..."),
+                    colour=Colours.soft_red,
+                    url=utils.RULES_URL
+                ).set_author(
+                    name=utils.INFRACTION_AUTHOR_NAME,
+                    url=utils.RULES_URL,
+                    icon_url=Icons.defcon_denied
+                ).set_footer(text=utils.INFRACTION_APPEAL_FOOTER),
+                "send_result": True
+            }
+        ]
+
+        for case in test_cases:
+            args = case["args"]
+            expected = case["expected_output"]
+            send = case["send_result"]
+
+            with self.subTest(args=args, expected=expected, send=send):
+                send_private_embed_mock.reset_mock()
+
+                send_private_embed_mock.return_value = send
+                result = await utils.notify_infraction(*args)
+
+                self.assertEqual(send, result)
+
+                embed = send_private_embed_mock.call_args[0][1]
+
+                self.assertEqual(embed.to_dict(), expected.to_dict())
+
+                send_private_embed_mock.assert_awaited_once_with(args[0], embed)
+
+    @patch("bot.cogs.moderation.utils.send_private_embed")
+    async def test_notify_pardon(self, send_private_embed_mock):
+        """Should send an embed of a certain format as a DM and return `True` if DM successful."""
+        test_cases = [
+            {
+                "args": (self.user, "Test title", "Example content"),
+                "icon": Icons.user_verified,
+                "send_result": True
+            },
+            {
+                "args": (self.user, "Test title", "Example content", Icons.user_update),
+                "icon": Icons.user_update,
+                "send_result": False
+            }
+        ]
+
+        for case in test_cases:
+            args = case["args"]
+            send = case["send_result"]
```

Comment:
More of an addition to my other comment about the choice to define cases using dicts. I think a simple tuple unpacked in the for loop would be fine:
```python
for args, icon, send_result in test_cases:
```
But it can be seen as a matter of personal preference. You're probably doing `args = case["args"]` for convenience further down the function, so I'm just trying to ask the question whether we can define the test cases such that they already are convenient to work with, without the intermediate step.

Reply:
I removed these when moved to `namedtuple`.